src/legacy/core_plugins/kbn_vislib_vis_types/public/types.ts

```diff
@@ -17,7 +17,31 @@
  * under the License.
  */
 
+export enum LegendPositions {
+  RIGHT = 'right',
+  LEFT = 'left',
+  TOP = 'top',
+  BOTTOM = 'bottom',
+}
 export interface CommonVislibParams {
   addTooltip: boolean;
-  legendPosition: 'right' | 'left' | 'top' | 'bottom';
+  legendPosition: LegendPositions;
+}
+
+export interface ValueAxis {
+  id: string;
```

Comment:
Could you please add other definitions here: `type`, `position` and so on, since we are trying to describe things fully

Reply:
Done.